### ISSUE TITLE
NullPointerException in ContentService#284

### ISSUE THREAD

**Issue by halfhp** [filestack/filepicker-android#103]
It appears that [`ContentService.getErrorType(RetrofitError)`line 284](https://github.com/Ink/filepicker-android/blob/master/filepicker-library/src/io/filepicker/services/ContentService.java#L284) assumes that the RetrofitError will always have a non-null response, for all exception types besides RetrofitError.Kind.NETWORK which isn't true.  I'm currently getting a NullPointerException as a result of this, which is causing our app to crash.

Ideally, some additional detail would be included describing the root cause of the error, such as a ref to the RetrofitError.

In any case, here's a quick tweak to getErrorType that would at least avoid the NPE:

```
public ApiErrorEvent.ErrorType getErrorType(RetrofitError error) {
        ApiErrorEvent.ErrorType errorType = null;

        if(error != null) {
            if(error.getKind().equals(RetrofitError.Kind.NETWORK)) {
                errorType = ApiErrorEvent.ErrorType.NETWORK;
            } else if(error.getResponse() != null && error.getResponse().getStatus() == 401){
                errorType = ApiErrorEvent.ErrorType.UNAUTHORIZED;
            }
        }

        return errorType;
}
```

**Comment by halfhp** [comment]
Update:  I wrote an instrumented version of Filepicker's ContentService with the above fix along with some exception tracking on the RetrofitError.  Turns out the root cause is this:

https://github.com/square/okhttp/issues/2628

I think it's still a good idea to implement the change above to prevent other similar issues, but in order to eliminate the root cause of the NPE, you should also uprev the Retrofit dependency.

**Comment by halfhp** [comment]
Update #2: Actually looks like the root cause is a bug in Filepicker.  This method in Utils.java:

```
public static String getUploadedFilename(String mimetype) {
        Date date = new Date();
        SimpleDateFormat ft = new SimpleDateFormat("dd MMMM yyyy hh:mm a");

        String basename = (mimetype.contains("video") ? "Video " : "Image ");
        String extension = (mimetype.contains("video") ? ".mp4" : ".jpg");
        return basename + ft.format(date) + extension;
}
```

Will create filenames with invalid chars in certain months when used with the FR locale, and likely others as well.  For example, this is a filename with an invalid char generated by Filepicker: Image **10 août 2016 06:13 AM.jpg**

When the library goes to upload the image, this filename is placed into a header by the method `FpApiClient.uploadFile(...)` and subsequently causes the issue.

**Comment by janhudzik** [comment]
Fixed in https://github.com/filestack/filepicker-android/commit/3c4a79e30e0162efc4b9100f16a159be080448a9. Will be available in 3.10.0 release.
